tests/proofs/dexter/dexter.md

```diff
@@ -201,6 +208,139 @@ As reference materials for understanding the contract intent, we will consult:
+We first define functions which build our parameter and our storage types.
+
+```k
+  syntax TypeName ::= #DexterParamType(Bool)                [function, functional]
+                    | #DexterVersionSpecificParamType(Bool) [function, functional]
+  // -----------------------------------------------------------------------------
+  rule #DexterParamType(IsFA2)
+    => (or
+          (or
+             (or
+                (or (pair address                        // addLiquidity
+                       pair nat
+                         pair nat timestamp)
+                    unit)                                // default
+                (or (pair address                        // removeLiquidity
+                       pair nat
+                         pair mutez
+                           pair nat timestamp)
+                    (pair option key_hash bool)))        // setBaker
+             (or
+                (or address                              // setLqtAddress
+                    address)                             // setManager
+                (or (pair address                        // tokenToToken
+                       pair nat
+                         pair address
+                           pair nat timestamp)
+                    (pair address                        // tokenToXtz
+                       pair nat
+                         pair mutez timestamp))))
+          (or
+             (or unit                                    // updateTokenPool
+                 #DexterVersionSpecificParamType(IsFA2)) // updateTokenPoolInternal
+             (pair address                               // xtzToToken
+                pair nat timestamp)))
+
+  rule #DexterVersionSpecificParamType(true)  => list pair pair address nat nat
+  rule #DexterVersionSpecificParamType(false) => nat
+
+  syntax TypeName ::= #DexterStorageType(Bool)                [function, functional]
+                    | #DexterVersionSpecificStorageType(Bool) [function, functional]
+  // -------------------------------------------------------------------------------
+  rule #DexterStorageType(IsFA2)
+    => pair nat
+         pair mutez
+           pair nat
+             pair bool
+               pair bool
+                 pair address
+                   pair address
+                     #DexterVersionSpecificStorageType(IsFA2)
+
+  rule #DexterVersionSpecificStorageType(true)  => pair nat address
+  rule #DexterVersionSpecificStorageType(false) => address
+```
+
+We also define a functions that serialize and deserialize our abstract parameters and state.
+
+```k
+  syntax Data ::= #LoadDexterParams(EntryPointParams) [function, functional]
+  // -----------------------------------------------------------------------
+  // FIXME
+
+  syntax KItem ::= #loadDexterState(Bool, EntryPointParams)
+  // ------------------------------------------------------
+  rule <k> #loadDexterState(IsFA2, Params)
+        => .K
+           ...
+       </k>
```

Comment:
```suggestion
  rule <k> #loadDexterState(IsFA2, Params) => . ... </k>
```

Two things here:

- `.` is considered the canonical unit. It's not conventional (following the `.Sort` convention), but it's what the unparser will pick.
- Let's not waste newlines, save some for the fishies.